src/str.rs

```diff
@@ -860,6 +865,38 @@ mod test {
         );
     }
 
+    #[test]
+    fn from_str_mantissa_overflow_1() {
+        // reminder:
+        assert_eq!(OVERFLOW_U96, 79228162514264337593543950336);
+        assert_eq!(
+            parse_str_radix_10("7922816251426433759354395033.56").unwrap().unpack(),
+            Decimal::from_i128_with_scale(7922816251426433759354395034, 0).unpack()
+        );
+        // This is a mantissa of OVERFLOW_U96 - 1 just before reaching the last digit.
+        // Previously, this would return Err("overflow from mantissa after rounding")
+        // instead of successfully rounding.
+    }
+
+    #[test]
+    fn from_str_mantissa_overflow_2() {
+        assert_eq!(
+            parse_str_radix_10("79228162514264337593543950335.6"),
+            Err(Error::from("Invalid decimal: overflow from mantissa after rounding"))
+        );
+        // this case wants to round to 79228162514264337593543950340. which is
```

Comment:
This is the interesting case. It does initially try to round to `79_228_162_514_264_337_593_543_950_336.`, but this mantissa is `>= OVERFLOW_U96`, so we hit the second iteration and try to round to `79_228_162_514_264_337_593_543_950_340.`, which is *also* `>= OVERFLOW_U96`. This makes me realize though that the rounding should iterate through the entire mantissa when we start with `scale=0`; this can round to `79_228_162_514_264_337_593_543_950_300.`.